### PR TITLE
feat(DDI-1134): add details component

### PR DESCRIPTION
**Pull request by chrisolsen**

### CODE REVIEW

**Comment by Radha-Alberta** [comment]
1. In safari browser, the yellow border doesn't have rounded corners
image
2. The hover and focus states doesn't have grey background as in Figma.
image
image
3. The chevron icon should align with the first line of heading
image

File: libs/web-components/src/common/styling.ts [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+  "5": "m",
+  "6": "l",
+  "7": "xl",
+  "8": "2xl",
+  "9": "3xl",
+  "10": "4xl",
+}
+
+/**
+ * Allow for 0-10 values to be used along side the existing Spacing values
```
Comment (ArakTaiRoth) [inline libs/web-components/src/common/styling.ts:18]: Do we want to document anywhere that this is a thing? I'm not sold on that it matters, just wondering if we allow it.

**Comment by ArakTaiRoth** [comment]
Feedback from Tom:

- [ ] icon is not changing to interactive-hover colour when closed and hovering
- [ ] background should be white when heading is open and not hovering
- [ ] text should be interactive-default when heading is open and not hovering
- [ ] Space between icon and heading should be 8px. Currently it is 2px+4px (margin on icon and heading respectively)
- [ ] There should be interior padding on the right side of the content when open (1rem)
- [ ] Total height should be 42px tall overall (aligns with buttons)

**Comment by Radha-Alberta** [comment]
Few other comments after checking with Tom.
1. The line height should be larger. It is currently 24px, and it should be 28px (--goa-line-height-2)
2. The content text should be styled as --goa-typography-body-m
3. The icon is also not vertically centered when the detail is open.
4. Currently, the focus on detail heading only has yellow border. But just like the tertiary button, on focus the background should be grey as well as the yellow border.

**Review by ArakTaiRoth — APPROVED** [review]

File: apps/angular-demo/src/app/detail/detail.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<h2>Detail</h2>
```
Comment (vanessatran-ddi) [inline apps/angular-demo/src/app/detail/detail.html:1]: Suggestion: Can we rename the file from detail.html to detail.component.html?

File: apps/angular-demo/src/app/detail/detail.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Component } from "@angular/core";
```
Comment (vanessatran-ddi) [inline apps/angular-demo/src/app/detail/detail.ts:1]: Suggestion: Can we rename detail.ts to detail.component.ts?

File: libs/docs/src/components/common/Details.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import { Meta, Story } from "@storybook/addon-docs";
+import {
+  Props,
+  Prop,
+  Tabs,
+  Tab,
+  CodeSnippet,
+  SupportInfo,
+  Markdown,
```
Comment (vanessatran-ddi) [inline libs/docs/src/components/common/Details.stories.mdx:9]: Picky: "Markdown" is un-used import.

**Review by vanessatran-ddi — APPROVED** [review]
I just have some comments. The rest is good.

**Pull request merged**